dwave/cloud/coders.py

```diff
@@ -278,52 +283,52 @@ def decode_qp_numpy(msg, return_matrix=True):
     return result
 
 
-def _encode_problem_as_bq_ref(problem):
-    assert isinstance(problem, str)
-
-    return problem
+def encode_problem_as_ref(problem):
+    """Encode the problem given via reference for submission in the `ref` data
+    format.
 
-def _encode_problem_as_bq_json(problem):
-    assert hasattr(problem, 'to_serializable')
+    Args:
+        problem (str):
+            A reference to an uploaded problem (Problem ID).
 
-    return problem.to_serializable(use_bytes=False)
+    Returns:
+        encoded submission dictionary
+    """
 
-def _encode_problem_as_bq_json_zlib(problem):
-    assert hasattr(problem, 'to_serializable')
+    if not isinstance(problem, str):
+        raise TypeError("unsupported problem type")
```

Comment:
This might confuse. If the reference is not a string it is likely a problem, and probably legit; maybe ""unsupported identifier format"?

Reply:
I've changed the exception message a bit, but either should be fine. This function should be considered *internal*, and its sole purpose is to create the `data` dictionary for submission to SAPI. It strictly requires a `str` argument, the problem ID.